generator/backend_generators/vllm_generator.py

```diff
@@ -0,0 +1,127 @@
+import gc
+import copy
+import random
+import os
+from time import time
+from typing import List
+
+import torch
+import tqdm
+from vllm import LLM, SamplingParams
+from vllm.distributed.parallel_state import destroy_model_parallel
+from loguru import logger
+
+
+class VLLMGenerator:
```

Comment:
if the generator backends are all supposed to be used interchangeably, would it make sense to let them inherit from a common abstract baseclass to more formaly define an interface?

Reply:
yes, that sounds like a good idea

Reply:
added base class as you suggested